test/uk/gov/hmrc/pillar2/generators/ModelGenerators.scala

```diff
@@ -516,9 +567,10 @@ trait ModelGenerators {
 
   val arbitraryWithIdRegDataFoLLP: Arbitrary[GrsResponse] = Arbitrary {
     for {
+      companyProfile                    <- arbitrary[CompanyProfile]
       partnershipEntityRegistrationData <- arbitrary[PartnershipEntityRegistrationData]
 
-    } yield GrsResponse(partnershipEntityRegistrationData = Some(partnershipEntityRegistrationData))
+    } yield GrsResponse(partnershipEntityRegistrationData = Some(partnershipEntityRegistrationData.copy(companyProfile = Some(companyProfile))))
```

Comment:
```
partnershipEntityRegistrationData <- arbitrary[PartnershipEntityRegistrationData].suchThat(_.companyProfile.isDefined)
```

Reply:
It is neater but I've grown to dislike the suchTaht method as it just spawns another generator that filters down the output, I tend to prefer to ensure EVERY item I generate matches what I want

Reply:
But for a simple case like this it's probably fine

Reply:
it's a style point. I'm happy for you to say you prefer it this way